authors/apps/authentication/backends.py

```diff
@@ -1,10 +1,54 @@
-# import jwt
-#
-# from django.conf import settings
-#
-# from rest_framework import authentication, exceptions
-#
-# from .models import User
+from django.conf import settings
+from django.http import HttpResponse
+from rest_framework.authentication import BaseAuthentication
+from rest_framework.authentication import get_authorization_header
+from rest_framework import status, exceptions
+import jwt
+from .models import User
 
-"""Configure JWT Here"""
 
+class JWTAuthentication(BaseAuthentication):
+    """This validates jwt token"""
+
+    def authenticate(self, request):
+        """Checks the token and verify its valid"""
+        auth = get_authorization_header(request).split()
+        if not auth or auth[0].lower() != b'token':
+            return None
+
+        if len(auth) != 2:
+            msg = 'Token header is invalid.'
+            raise exceptions.AuthenticationFailed(msg)
+
+        try:
+            token = auth[1]
+            if token == "null":
+                msg = 'Null token not allowed'
+                raise exceptions.AuthenticationFailed(msg)
+        except UnicodeError:
+            msg = 'Invalid token header. Token string contains invalid characters.'
+            raise exceptions.AuthenticationFailed(msg)
+
+        return self.authenticate_credentials(token)
+
+    def authenticate_credentials(self, token):
+        """Check that the user associated with the token
+        exists in our database"""
+        payload = jwt.decode(token, settings.SECRET_KEY)
+        email = payload['email']
+        userid = payload['id']
+        msg = {'Error': "Token mismatch", 'status': status.HTTP_401_UNAUTHORIZED}
```

Comment:
Where are you using this msg?

Reply:
We replaced that msg with "user does not exist", and forgot to get rid of it. Let me fix that.